adapter.go

```diff
@@ -254,8 +255,26 @@ func NewAdapterByDBWithCustomTable(db *gorm.DB, t interface{}) (*Adapter, error)
 	}
 
 	ctx = context.WithValue(ctx, customTableKey{}, t)
+	tableName := camelToSnakeString(reflect.TypeOf(t).Elem().Name())
 
-	return NewAdapterByDBUseTableName(db.WithContext(ctx), "", defaultTableName)
+	return NewAdapterByDBUseTableName(db.WithContext(ctx), "", tableName)
```

Comment:
How about just modify the method by add a `...string` parameter called `tablename`, it is easy to use and easy to do.